src/Ultz.DF2/CommandSender.cs

```diff
@@ -80,45 +80,14 @@ public void SendValue(string name, object value, out ValueKind valueKind)
                 throw new ArgumentNullException(nameof(name));
             }
 
-            if (value is not byte &&
-                value is not sbyte &&
-                value is not short &&
-                value is not ushort &&
-                value is not int &&
-                value is not uint &&
-                value is not long &&
-                value is not ulong &&
-                value is not float &&
-                value is not double &&
-                value is not string &&
-                value is not byte[] &&
-                value is not sbyte[] &&
-                value is not short[] &&
-                value is not ushort[] &&
-                value is not int[] &&
-                value is not uint[] &&
-                value is not long[] &&
-                value is not ulong[] &&
-                value is not float[] &&
-                value is not double[] &&
-                value is not string[] &&
-                value is not IEnumerable &&
-                value is not null)
-            {
-                throw new ArgumentException(
-                    "Value is of an invalid type. Valid types are (or inherit from) byte, sbyte, short, ushort, int, " +
-                    "uint, long, ulong, float, double, string, an array of one of the aforementioned types, or " +
-                    "IEnumerable");
-            }
```

Comment:
How come this isn't needed? Do we do this check elsewhere?

Reply:
Implicitly via the if-else chain as far as I can see. I added it to the `else` at the end. This puts the burden of testing for every value on only bad code